### PR TITLE
edh limit max inputs cherry pick

### PR DESCRIPTION
**Pull request by scottmillner**
supercedes: https://github.com/botanix-labs/Macbeth/pull/311

### CODE REVIEW

File: bin/btc-server/src/signer.rs [hunk 1]
```diff
@@ -298,7 +298,8 @@ impl App {
             reth_btc_wallet::address::generate_taproot_change_scriptpubkey(&secp_pk);
         let mut original_psbt = self
             .make_tx(outputs, fee_rate, change_script.clone(), checkpoint_block, utxo_merkle_root)
-            .await?;
+            .await
+            .unwrap();
```
Comment (arminsabouri) [inline bin/btc-server/src/signer.rs]: Unwrap seems strictly worse

File: crates/consensus/auto-seal/Cargo.toml [hunk 2]
```diff
@@ -32,7 +32,7 @@ futures-util.workspace = true
 tokio = { workspace = true, features = ["sync", "time"] }
 tokio-stream.workspace = true
 tracing.workspace = true
-
+bitcoin.workspace = true
```
Comment (arminsabouri) [inline crates/consensus/auto-seal/Cargo.toml]: Is this change related? Or just fixing the build

File: crates/botanix-lib/src/peg_contract.rs [hunk 3]
```diff
@@ -433,6 +431,50 @@ mod tests {
             meta: vec![meta],
         }
     }
+}
+
+#[cfg(test)]
+pub mod tests {
```
Comment (arminsabouri) [inline crates/botanix-lib/src/peg_contract.rs]: Did we accidentally remove these tests?

**Review by arminsabouri — CHANGES_REQUESTED** [review]
Had one question about an unwrap. And a couple other things

File: bin/btc-server/src/signer.rs (hunk 1 above)
Comment (scottmillner) [reply bin/btc-server/src/signer.rs]: agreed, rolledback

File: crates/consensus/auto-seal/Cargo.toml (hunk 2 above)
Comment (scottmillner) [reply crates/consensus/auto-seal/Cargo.toml]: not related nor needed...rolledback

File: crates/botanix-lib/src/peg_contract.rs (hunk 3 above)
Comment (scottmillner) [reply crates/botanix-lib/src/peg_contract.rs]: mistake...it included two tests twice...rolledback

**Review by arminsabouri — APPROVED** [review]
CodeACk
Just squash the last two commits and we are good to go

**Pull request merged**